### DISCUSSION TITLE
Loading full parent object according to nested relations

### DISCUSSION THREAD

**Discussion by heshamz** [mikro-orm/mikro-orm#2125] [General]
Hi All,

I am facing a strange issue, I have the following entities, and I want to search the parent according to children, the issue is after searching not all the children have been loaded, only according to the criteria, but I want to load the full part object according to criteria without affecting nested relations.

Could you please support me on this.

**Entities**
```
@Entity()
export class MeasureUnit  {
  @Property()
  name!: string;

  @Property()
  symbol: string;

  @OneToMany(
    () => MeasureUnitTranslatableContent,
    (t: MeasureUnitTranslatableContent) => t.parent,
    {
      cascade: [Cascade.ALL],
      eager: true,
      orphanRemoval: true,
    }
  )
  translations = new Collection<MeasureUnitTranslatableContent>(this);
}

@Entity()
export class MeasureUnitTranslatableContent extends TranslatableContent {
  @ManyToOne()
  parent: MeasureUnit;
}
```

**Search** 

```
const result = await this.entityRepository.find(
      {
        translations: {
          property: propertyParam,
          locale: localeParam,
          contents: { $like: contentsParam },
        },
      },
      { refresh: true, cache: false }
    );
```

**Generated SQL**

```
[query] select `e0`.* from `measure_unit` as `e0` left join `measure_unit_translatable_content` as `e1` on `e0`.`id` = `e1`.`parent_id` where `e1`.`property` = 'name' and `e1`.`locale` = 'en' and `e1`.`contents` like 'Gra%' [took 0 ms]
[query] select `e0`.* from `measure_unit_translatable_content` as `e0` where `e0`.`parent_id` in (1) and `e0`.`property` = 'name' and `e0`.`locale` = 'en' and `e0`.`contents` like 'Gra%' order by `e0`.`parent_id` asc [took 0 ms]
```

**Comment by B4nan** [comment]
You need to do it separately, first the parent entity, then separate `em.populate()` call to load the relations - but this time without the query condition. Also note that result cache is disabled by default and you should usually not need `refresh` either. 

So drop the `eager: true` from the relation first, then:

```ts
const result = await this.entityRepository.find({
  translations: {
    property: propertyParam,
    locale: localeParam,
    contents: { $like: contentsParam },
  },
});
await this.entityRepository.populate(result, ['translations']);
```

**Reply by heshamz** [reply]
Thanks B4nan, actually just removing the `eager: true` and keeping everything else the same worked fine.
Apparently, the eager parameter prevents the populate to work normally.

**Reply by B4nan** [reply]
Well, without eager, you need to populate it explicitly, it wont be loaded on its own. Eager basically adds the property to populate list automatically, it does not prevent anything, it works exactly as I did describe it above.

**Reply by heshamz** [reply]
But I didn't use populate, I just removed the eager from the entity and kept everything else the same, and it worked.
